src/util/replaygain.h

```diff
@@ -56,6 +69,18 @@ class ReplayGain {
         m_peak = CSAMPLE_PEAK;
     }
 
+    // Parse and format the peak value metadata according to the ReplayGain
+    // 1.0/2.0 specification.
+    // http://wiki.hydrogenaud.io/index.php?title=ReplayGain_1.0_specification
+    // http://wiki.hydrogenaud.io/index.php?title=ReplayGain_2.0_specification
+    static CSAMPLE parsePeak(QString strPeak, bool* pValid = 0);
```

Comment:
I also thought about adding a static assert, because we implicitly assume 
here that CSAMPLE_PEAK = 1.0

On 11/03/2015 08:15 AM, Daniel Schürmann wrote:
>
> In src/util/replaygain.h 
> <https://github.com/mixxxdj/mixxx/pull/766#discussion_r43721107>:
>
> > @@ -56,6 +69,18 @@ class ReplayGain {
> >          m_peak = CSAMPLE_PEAK;
> >      }
> >
> > +    // Parse and format the peak value metadata according to the ReplayGain
> > +    // 1.0/2.0 specification.
> > +    // http://wiki.hydrogenaud.io/index.php?title=ReplayGain_1.0_specification
> > +    // http://wiki.hydrogenaud.io/index.php?title=ReplayGain_2.0_specification
> > +    static CSAMPLE parsePeak(QString strPeak, bool* pValid = 0);
>
> An additional hint that strPeak = "1.0" represents the digital full scale 
> could be useful.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mixxxdj/mixxx/pull/766/files#r43721107>.
>